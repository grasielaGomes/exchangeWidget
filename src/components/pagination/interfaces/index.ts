export interface PaginationI {
  handlePage: (page: number) => void;
}
