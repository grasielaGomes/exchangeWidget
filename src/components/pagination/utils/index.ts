export const getNumbers = (number: number): string[] => {
  const array = [];
  for (let i = 1; i <= number; i++) {
    array.push(String(i));
  }
  return array;
};
