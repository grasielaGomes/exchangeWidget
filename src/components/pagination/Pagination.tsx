import { TextButton } from "../buttons/TextButton";
import { NextArrow } from "../icons/NextArrow";
import { usePagination } from "./hooks/usePagination";
import { useEffect } from "react";
import { PaginationI } from "./interfaces";

const styles = {
  container: "mt-7 gap-1 hidden md:flex",
  arrowIcon: "fill-dark"
};

export const Pagination = ({ handlePage }: PaginationI) => {
  const {
    counter,
    currentPage,
    handleCurrentPage,
    handleLastPage,
    handleNextPage,
    handlePlusButton,
    isBigList,
    pages,
    showPages
  } = usePagination();

  useEffect(() => {
    handlePage(Number(currentPage));
  }, [currentPage]);

  const showNextButton =
    isBigList && counter < pages.length && Number(currentPage) !== pages.length;

  return (
    <div className={styles.container}>
      {showPages.map((page) => (
        <TextButton
          key={page}
          handleClick={() => {
            handleCurrentPage(page);
          }}
          isActive={page === currentPage}
        >
          {page}
        </TextButton>
      ))}
      {showNextButton && (
        <>
          <TextButton handleClick={handlePlusButton}>...</TextButton>
          <TextButton
            handleClick={handleLastPage}
            isActive={pages.slice(-1)[0] === currentPage}
          >
            {pages.slice(-1)[0]}
          </TextButton>
          <TextButton handleClick={handleNextPage} variant="icon">
            <span>Next</span>
            <NextArrow className={styles.arrowIcon} />
          </TextButton>
        </>
      )}
    </div>
  );
};
