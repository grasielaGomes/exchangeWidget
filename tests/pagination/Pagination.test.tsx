import { render, screen } from "@testing-library/react";
import { Pagination } from "../../src/components/pagination/Pagination";

describe("Tests on <Pagination />", () => {
  const transactions = [
    {
      amount: "100",
      currencyRate: 48000,
      date: "2020-01-01 08:02:17",
      from: "Bitcoin",
      id: "1",
      to: "USD",
      totalAmount: "200",
      status: "LIVE"
    },
    {
      amount: "100",
      currencyRate: 3.45,
      date: "2020-01-01 08:02:17",
      from: "Ethereum",
      id: "2",
      to: "EUR",
      totalAmount: "145",
      status: "EXCHANGED"
    }
  ];
  const { container } = render(
    <Pagination handlePage={() => {}} list={transactions} />
  );

  it("should match with snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should show 1 page", () => {
    render(<Pagination handlePage={() => {}} list={transactions} />);
    expect(screen.getByText("1")).toBeTruthy();
  });

  it("should show custom error message", () => {});
});
