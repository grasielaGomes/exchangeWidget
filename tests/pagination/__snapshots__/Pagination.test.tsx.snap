// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Pagination /> should match with snapshot 1`] = `
<div>
  <div
    class="mt-7 gap-1 hidden md:flex"
  >
    <button
      class="transition-colors duration-200 ease-in-out rounded-lg bg-transparent text-base leading-4 underline-offset-1 disabled:bg-neutral disabled:hover:bg-neutral focus:outline-none focus:border focus:border-dark focus:ring-dark focus:ring-0 bg-dark text-white h-10 w-10"
      type="button"
    >
      1
    </button>
  </div>
</div>
`;
